.github/scripts/changed-modules.sh

```
@@ -0,0 +1,84 @@
+# On CI, this value will come from a Github Action retrieving the list of modified files from the pull request.
+readonly modified_files=${ALL_CHANGED_FILES[@]}
+
+# Initialize variables
+modified_modules=()
+
+# Check the modified files and determine which modules to build, following these rules:
+# - if the modified files only contain files in one of the modules, include that module in the list
+# - if the modified files contain any other file, include all modules in the list
+for file in $modified_files; do
+    if [[ $file == .github/* ]]; then
+        modified_modules=${allModules[@]}
```

Shellcheck (0.10.0)

[warning] Variable was used as an array but is now assigned a string.

(SC2178)